src/main/java/cloud/orbit/actors/extensions/metrics/dropwizard/OrbitActorExtension.java

```diff
@@ -137,40 +141,53 @@ public void onRead(HandlerContext ctx, Object message)
-        Counter counter = MetricsManager.getInstance().getRegistry().counter(actorCounterKey);
-        counter.dec();
+        MetricsManager.getInstance().getRegistry().counter(actorCounterKey).dec();
 
-        //actor lifespan
-        String startTimeKey = getActorStartTimeKey(actor);
+        //actor deactivation
+        String actorDeactivationTimerKey = getDeactorActivationTimerKey(actor);
+        Timer.Context deactivationTimer = actorDeactivationTimers.remove(actorDeactivationTimerKey);
+        if (null != deactivationTimer)
+        {
+            deactivationTimer.stop();
+        }
 
-        Long startTime = actorStartTimes.get(startTimeKey);
-        if (null != startTime)
+        //Actor lifespan
+        String actorLifespanTimerKey = getActorLifespanTimerKey(actor);
+        String actorTypeLifespanMetricsKey = getActorLifeSpanMetricsKey(RemoteReference.getInterfaceClass(actor));
+        Timer.Context lifespanTimer = actorLifeSpanTimers.remove(actorLifespanTimerKey);
+        if (null != lifespanTimer)
         {
-            actorStartTimes.remove(startTimeKey);
-
-            String actorLifeSpanHistogramKey = getActorLifeSpanHistogramMetricsKey(RemoteReference.getInterfaceClass(actor));
-            Histogram histogram = actorLifeSpanHistograms.get(actorLifeSpanHistogramKey);
-            if (null != histogram)
-            {
-                histogram.update(System.currentTimeMillis() - startTime);
-            }
+            lifespanTimer.stop();
         }
+
         return Task.done();
     }
 
-    public static String getActorStartTimeKey(final AbstractActor<?> actor)
+    public static String getActorActivationTimerKey(final AbstractActor<?> actor)
+    {
+        Object id = RemoteReference.getId(actor);
+        return String.format("actors.%s.%s.activation.timer", RemoteReference.getInterfaceClass(actor).getSimpleName(), id);
+    }
+
+    public static String getDeactorActivationTimerKey(final AbstractActor<?> actor)
     {
         Object id = RemoteReference.getId(actor);
-        return String.format("actors.%s.%s.starttime", RemoteReference.getInterfaceClass(actor).getSimpleName(), id);
+        return String.format("actors.%s.%s.deactivation.timer", RemoteReference.getInterfaceClass(actor).getSimpleName(), id);
+    }
+
+    public static String getActorLifespanTimerKey(final AbstractActor<?> actor)
+    {
+        Object id = RemoteReference.getId(actor);
+        return String.format("actors.%s.%s.lifespan.timer", RemoteReference.getInterfaceClass(actor).getSimpleName(), id);
     }
 
     /**
-     * Metric key: Actor message receive rate
+     * Metrics key: Actor message receive rate
      *
      * @param actorClass
      * @return
      */
-    public static String getActorMessageReceiveRateMetricKey(Class<? extends Actor> actorClass)
+    public static String getActorMessageReceiveRateMetricsKey(Class<? extends Actor> actorClass)
```

Comment:
For consistency include the word "Type"... getActorTypeMessageReceiveRateMetricsKey().